src/Bridge/Symfony/Bundle/DependencyInjection/Compiler/MetadataAwareNameConverterPass.php

```diff
@@ -33,14 +34,20 @@ final class MetadataAwareNameConverterPass implements CompilerPassInterface
      */
     public function process(ContainerBuilder $container)
     {
-        if ($container->hasAlias('api_platform.name_converter') || !$container->hasDefinition('serializer.name_converter.metadata_aware')) {
+        if (!$container->hasDefinition('serializer.name_converter.metadata_aware')) {
             return;
         }
 
         $definition = $container->getDefinition('serializer.name_converter.metadata_aware');
-
-        if (1 >= \count($definition->getArguments()) || null === $definition->getArgument(1)) {
-            return;
+        $num = \count($definition->getArguments());
+
+        if ($container->hasAlias('api_platform.name_converter')) {
+            $nameConverter = new Reference((string) $container->getAlias('api_platform.name_converter'));
```

Comment:
```suggestion
            $nameConverter = new Reference('api_platform.name_converter');
```

(Not sure if possible to create a `Reference` using an alias id.)

Reply:
I have a circular dependency if I do this.